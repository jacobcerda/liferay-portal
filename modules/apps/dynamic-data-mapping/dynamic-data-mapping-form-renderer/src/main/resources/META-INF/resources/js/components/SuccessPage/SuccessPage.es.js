/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import Component from 'metal-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import {setValue} from '../../util/i18n.es';
import templates from './SuccessPage.soy';

class SuccessPage extends Component {
	prepareStateForRender(state) {
		const {editingLanguageId, successPageSettings} = this;
		const {body, title} = successPageSettings;

		if (body[editingLanguageId] == '') {
			body[editingLanguageId] = Liferay.Language.get(
				'your-information-was-successfully-received-thank-you-for-filling-out-the-form'
			);
		}

		if (title[editingLanguageId] == '') {
			title[editingLanguageId] = Liferay.Language.get('thank-you');
		}

		return {
			...state,
			body: (body && body[editingLanguageId]) || '',
			enabled: true,
			title: (title && title[editingLanguageId]) || '',
		};
	}

	shouldUpdate(changes) {
		const {editingLanguageId} = changes;

		return (
			editingLanguageId &&
			editingLanguageId.newVal !== editingLanguageId.prevVal
		);
	}

	_handleSuccessPageUpdated(event) {
		const {dispatch, store} = this.context;
		const {editingLanguageId} = store.props;
		const {delegateTarget} = event;
		const {
			dataset: {setting},
			value,
		} = delegateTarget;
		const {successPageSettings} = this;

		dispatch(
			'successPageChanged',
			setValue(successPageSettings, editingLanguageId, setting, value)
		);
	}
}

SuccessPage.STATE = {
	editingLanguageId: Config.string(),

	strings: Config.object().value({
		'success-page': Liferay.Language.get('success-page'),
	}),

	/**
	 * @instance
	 * @memberof SuccessPage
	 * @type {?object}
	 */

	successPageSettings: Config.object().value({
		body: {},
		title: {},
	}),
};

Soy.register(SuccessPage, templates);

export default SuccessPage;
