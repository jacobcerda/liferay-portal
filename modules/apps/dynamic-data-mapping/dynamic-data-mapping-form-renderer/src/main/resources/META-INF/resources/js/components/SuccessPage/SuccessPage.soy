{namespace SuccessPage}

{template .render}
	{@param? _handleSuccessPageUpdated: any}
	{@param? body: string}
	{@param? strings: map<string, string>}
	{@param? title: string}

	{delcall SuccessPage.container}
		{param _handleSuccessPageUpdated: $_handleSuccessPageUpdated /}
		{param body: $body /}
		{param strings: $strings /}
		{param title: $title /}
	{/delcall}
{/template}

{deltemplate SuccessPage.container}
	{@param? _handleSuccessPageUpdated: any}
	{@param? body: string}
	{@param? strings: map<string, string>}
	{@param? title: string}

	<div class="sheet simple-page ddm-page-success-layout">
		<div class="form-builder-layout">
			<h5 class="pagination">{$strings != null ? $strings['success-page'] : ''}</h5>

			<div class="active ddm-form-page lfr-ddm-form-page form-builder-success-page">

				<input
					class="form-builder-page-header-title form-control p-0"
					maxlength="120"
					placeholder=""
					data-oninput="{$_handleSuccessPageUpdated}"
					data-setting="title"
					type="text"
					value="{$title}"
				/>

				<input
					class="form-builder-page-header-description form-control p-0"
					data-oninput="{$_handleSuccessPageUpdated}"
					data-setting="body"
					type="text"
					maxlength="120"
					placeholder=""
					value="{$body}"
				/>
			</div>
		</div>
	</div>
{/deltemplate}