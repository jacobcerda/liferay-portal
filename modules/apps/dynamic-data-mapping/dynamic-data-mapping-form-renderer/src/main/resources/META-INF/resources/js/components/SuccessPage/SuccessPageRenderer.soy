{namespace SuccessPageRenderer}

{deltemplate PageRenderer.container variant="'success'"}
	{@param activePage: number}
	{@param cancelLabel: string}
	{@param pageIndex: number}
	{@param pages: ?}
	{@param showCancelButton: bool}
	{@param showSubmitButton: bool}
	{@param spritemap: string}
	{@param submitLabel: string}
	{@param total: number}
	{@param? _handleFieldBlurred: any}
	{@param? _handleFieldClicked: any}
	{@param? _handleFieldEdited: any}
	{@param? editable: bool}
	{@param? editingLanguageId: string}
	{@param? empty: bool}
	{@param? page: ?}
	{@param? paginationMode: ?}
	{@param? readOnly: bool}
	{@param? redirectURL: string}
	{@param? strings: map<string, string>}

		{call SuccessPage.render}
			{param editingLanguageId: $editingLanguageId /}
			{param successPageSettings: $page?.successPageSettings /}
		{/call}
{/deltemplate}