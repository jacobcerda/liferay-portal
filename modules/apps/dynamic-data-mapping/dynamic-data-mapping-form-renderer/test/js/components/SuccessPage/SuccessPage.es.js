/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import dom from 'metal-dom';

import SuccessPage from '../../../../src/main/resources/META-INF/resources/js/components/SuccessPage/SuccessPage.es';
import SuccessPageSettings from '../../__mock__/mockSuccessPage.es';
import withContextMock from '../../__mock__/withContextMock.es';

let component;
let successPageSettings;
const SuccessPageWithContextMock = withContextMock(SuccessPage);

const createSuccessPage = (props) => {
	return new SuccessPageWithContextMock({
		contentLabel: 'Content',
		editingLanguageId: 'en_US',
		successPageSettings,
		titleLabel: 'Title',
		...props,
	});
};

describe('SuccessPage', () => {
	beforeEach(() => {
		successPageSettings = JSON.parse(JSON.stringify(SuccessPageSettings));

		jest.useFakeTimers();
	});

	afterEach(() => {
		if (component) {
			component.dispose();
		}

		successPageSettings = null;
	});

	it('renders the component', () => {
		component = createSuccessPage();

		jest.runAllTimers();

		expect(component).toMatchSnapshot();
	});

	it('emits event when page title is changed', () => {
		component = createSuccessPage();

		const titleNode = component.element.querySelector(
			'input[data-setting="title"]'
		);

		titleNode.value = 'Some title';
		dom.triggerEvent(titleNode, 'input', {});

		expect(component.context.dispatch).toHaveBeenCalledWith(
			'successPageChanged',
			{
				body: {
					en_US:
						'your-information-was-successfully-received-thank-you-for-filling-out-the-form',
				},
				enabled: true,
				title: {
					en_US: 'Some title',
				},
			}
		);
	});

	it('emits event when page body is changed', () => {
		component = createSuccessPage();

		const bodyNode = component.element.querySelector(
			'input[data-setting="body"]'
		);

		bodyNode.value = 'Some description';
		dom.triggerEvent(bodyNode, 'input', {});

		expect(component.context.dispatch).toHaveBeenCalledWith(
			'successPageChanged',
			{
				body: {
					en_US: 'Some description',
				},
				enabled: true,
				title: {
					en_US: 'thank-you',
				},
			}
		);
	});
});
