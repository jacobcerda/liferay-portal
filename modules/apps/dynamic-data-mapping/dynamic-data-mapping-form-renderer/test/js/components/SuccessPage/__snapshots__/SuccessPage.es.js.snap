// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessPage renders the component 1`] = `
<div class="sheet simple-page ddm-page-success-layout">
  <div class="form-builder-layout">
    <h5 class="pagination">success-page</h5>
    <div class="active ddm-form-page lfr-ddm-form-page form-builder-success-page"><input class="form-builder-page-header-title form-control p-0" maxlength="120" placeholder="" data-setting="title" type="text" value="thank-you"><input class="form-builder-page-header-description form-control p-0" data-setting="body" type="text" maxlength="120" placeholder="" value="your-information-was-successfully-received-thank-you-for-filling-out-the-form"></div>
  </div>
</div>
`;
